### PR TITLE
WRP-6257: Cell: Add `grow` prop to expand the size to its container

### PR DESCRIPTION
**Pull request by hoeunryuLG**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [x] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
A `grow`able cell will expand to its maximum size according to the remaining space of the container. This is used when you want to grow the size of this cell so that it fills the container.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
add `grow` prop to `Cell` component.

A `grow`able Cell has a declaration of `flex: 1 1 100%` in CSS by default. However, the `flex-basis` property is overridden by the `var(--cell-size)` value passed from JS. And by combining `max-width: none` and `flex-grow: 1` declaration, its size can be increased by the size of its parent container.

Also, by the declaration of `flex: 1 1 100%`, the `flex-shrink` property has a value of `1`, which means that its size can be reduced as much as necessary, but because of `min-width: var(--cell-size)` and `min-height: var(--cell-size)` declaration, it does not decrease below the specified cell size.

When combined with `shrink`, `shrink` prop takes precedence over `grow` prop and `grow` prop is simply ignored. When used in conjunction with `size`, the size will be the minimum size, growing as necessary, to fit the container.

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)
Screenshot tests are added to Sandstone (enactjs/sandstone#1417)

Unit Test Cases
- `grow` CSS class is applied when no props (`shrink`, `size`) are  given
- `grow` CSS class and `flex-basis` CSS style are applied when `grow` and `size` props are given
- `grow` CSS class is not applied when used with `shrink` prop

Sampler
- The thrid Cell is modified to change the `grow` and `size` props.The part where `grow` is automatically activated is still checkable when `shrink` prop is disabled in the second Cell.
- Clarify the names of each Cell and control to make them easier to distinguish.

### Links
[//]: # (Related issues, references)
WRP-6257
enactjs/sandstone#1417

### Comments
Enact-DCO-1.0-Signed-off-by: Hoeun Ryu <hoeun.ryu@lge.com>

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,11 @@
 
 The following is a curated list of changes in the Enact project, newest changes on the top.
 
+## [unreleased]
+
+### Added
+- `ui/Cell` prop `grow` to expand its size to the container
```
Comment (MikyungKim) [inline CHANGELOG.md:8]: If you are making a change of ui package, please add a change log under packages/ui/CHANGELOG.md

Comment (MikyungKim) [reply CHANGELOG.md:8]: Also, the module name would be `ui/Layout.Cell`.

File: packages/ui/Layout/Cell.js [hunk 2]
```diff
@@ -109,6 +130,7 @@ const CellBase = kind({
 
 	defaultProps: {
 		component: 'div',
+		grow: false,
```
Comment (MikyungKim) [inline packages/ui/Layout/Cell.js:133]: We don't need to specify false default props.

File: packages/ui/Layout/Cell.js [hunk 3]
```diff
@@ -70,6 +70,21 @@ const CellBase = kind({
...
+		 * container. This is used when you want to grow the size of this Cell so that it fills the
+		 * container. See the {@link ui/Layout.Cell#size|size} property for more details.
+		 *
+		 * When combined with {@link ui/Layout.Cell#shrink|shrink}, `shrink` prop takes precedence over
+		 * `grow` prop and `grow` prop is simply ignored.
+		 *
+		 * @type {Boolean}
+		 * @public
+		 */
+		grow: PropTypes.bool,
```
Comment (MikyungKim) [inline packages/ui/Layout/Cell.js:86]: Would you please add unit tests for this prop?

File: CHANGELOG.md (hunk 1 above)
Comment (hoeunryuLG) [reply CHANGELOG.md:8]: OK. I will fix it.

File: packages/ui/Layout/Cell.js (hunk 2 above)
Comment (hoeunryuLG) [reply packages/ui/Layout/Cell.js:133]: I will fix it.

File: packages/ui/Layout/Cell.js (hunk 3 above)
Comment (hoeunryuLG) [reply packages/ui/Layout/Cell.js:86]: OK. I will add unit tests for this prop.

**Comment by hoeunryuLG** [comment]
A new patch set has been pushed to address reviewer's comments.

File: packages/ui/CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,11 @@
 
 The following is a curated list of changes in the Enact ui module, newest changes on the top.
 
+## [unreleased]
+
+### Added
+- `ui/Layout.Cell` prop `grow` to expand its size to the container
```
Comment (MikyungKim) [inline packages/ui/CHANGELOG.md:8]:
```suggestion

- `ui/Layout.Cell` prop `grow` to expand its size to the container
```

File: packages/sampler/stories/default/Layout.js [hunk 5]
```diff
@@ -29,7 +29,10 @@ export const _Layout = (args) => (
 			<Cell shrink={args['shrinkable cell']}>
 				<Item>Second</Item>
 			</Cell>
-			<Cell>
+			<Cell
+				grow={args['growable cell']}
```
Comment (MikyungKim) [inline packages/sampler/stories/default/Layout.js:33]: It's quite ambiguous that `grow` control matches this Cell. Let's rename it clearer. Please do the same to `shrink`

File: packages/sampler/stories/default/Layout.js [hunk 6]
```diff
@@ -29,7 +29,10 @@ export const _Layout = (args) => (
...
+				size={args['cell size'] + 'px'}
```
Comment (MikyungKim) [inline packages/sampler/stories/default/Layout.js:34]: Let's separate the args for the first cell size and this one.

Comment (hoeunryuLG) [reply packages/sampler/stories/default/Layout.js]: I'll fix it.

File: packages/sampler/stories/default/Layout.js (hunk 5 above)
Comment (hoeunryuLG) [reply packages/sampler/stories/default/Layout.js]: I'll fix it.

File: packages/ui/CHANGELOG.md (hunk 4 above)
Comment (hoeunryuLG) [reply packages/ui/CHANGELOG.md:8]: I'll fix it.

**Comment by hoeunryuLG** [comment]
A new patch set has been pushed and PR description has been changed to address reviewer's comments.

File: packages/sampler/stories/default/Layout.js [hunk 7]
```diff
@@ -21,28 +21,33 @@ export const _Layout = (args) => (
...
 			</Cell>
-			<Cell shrink={args['shrinkable cell']}>
-				<Item>Second</Item>
+			<Cell shrink={args['shrinkable second cell']}>
+				<Item>Second (shrinkable)</Item>
 			</Cell>
-			<Cell>
-				<Item>Third</Item>
+			<Cell
+				grow={args['growable thrid cell']}
```
Comment (MikyungKim) [inline packages/sampler/stories/default/Layout.js:33]: oops, typo. third -> third

File: packages/sampler/stories/default/Layout.js [hunk 8]
```diff
@@ -21,28 +21,33 @@ export const _Layout = (args) => (
...
 );
 
 select('align', _Layout, ['start', 'center', 'stretch', 'end'], Layout, 'start');
 select('orientation', _Layout, ['horizontal', 'vertical'], Layout, 'horizontal');
-range('cell size', _Layout, Cell, {min: 0, max: 300, step: 5}, 100);
-boolean('shrinkable cell', _Layout, Cell);
+range('first cell size', _Layout, Cell, {min: 0, max: 300, step: 5}, 100);
+boolean('shrinkable second cell', _Layout, Cell);
+range('third cell size', _Layout, Cell, {min: 0, max: 600, step: 5}, 120);
+boolean('growable thrid cell', _Layout, Cell);
```
Comment (MikyungKim) [inline packages/sampler/stories/default/Layout.js:50]: typo

File: packages/sampler/stories/default/Layout.js [hunk 9]
```diff
@@ -21,28 +21,33 @@ export const _Layout = (args) => (
 			orientation={args['orientation']}
 		>
 			<Cell
-				size={args['cell size'] + 'px'}
+				size={args['first cell size'] + 'px'}
 				shrink
 			>
-				<Item>First</Item>
+				<Item>First (shrink)</Item>
```
Comment (MikyungKim) [inline packages/sampler/stories/default/Layout.js:27]: I'm not sure but it seems that adding "(shrink)" affects its height when the size is small.
🤔 Should we change the content?

File: packages/ui/CHANGELOG.md (hunk 4 above)
Comment (MikyungKim) [reply packages/ui/CHANGELOG.md:8]: Please fix it.

File: packages/sampler/stories/default/Layout.js (hunk 7 above)
Comment (hoeunryuLG) [reply packages/sampler/stories/default/Layout.js:33]: I'll fix it.

File: packages/sampler/stories/default/Layout.js (hunk 8 above)
Comment (hoeunryuLG) [reply packages/sampler/stories/default/Layout.js:50]: I'll fix it

File: packages/sampler/stories/default/Layout.js (hunk 9 above)
Comment (hoeunryuLG) [reply packages/sampler/stories/default/Layout.js:27]: I'll add `NO-BREAK SPACE`, which is &#xa0; between First and (shrink).

**Comment by hoeunryuLG** [comment]
PR has been updated to address comments.
1. merge with the latest develop branch
2. fix the typo, thrid -> third
3. use `NO-BREAK SPACE` instead of normal space for each Cell's content in the sampler.

File: packages/ui/CHANGELOG.md (hunk 4 above)
Comment (hoeunryuLG) [reply packages/ui/CHANGELOG.md:9]: it's been fixed.

**Review by MikyungKim — APPROVED** [review]
Nicely done! :lgtm:

File: packages/ui/Layout/Cell.js [hunk 10]
```diff
@@ -91,10 +106,16 @@ const CellBase = kind({
 		 * When used in conjunction with {@link ui/Layout.Cell#shrink|shrink}, the size will be
 		 * the maximum size, shrinking as necessary, to fit the content.
 		 *
+		 * When used in conjunction with {@link ui/Layout.Cell#grow|grow}, the size will be the
```
Comment (seunghoh) [inline packages/ui/Layout/Cell.js:109]: This link seems not working.

File: packages/ui/Layout/Cell.js [hunk 11]
```diff
@@ -91,10 +106,16 @@ const CellBase = kind({
 		 * When used in conjunction with {@link ui/Layout.Cell#shrink|shrink}, the size will be
```
Comment (seunghoh) [inline packages/ui/Layout/Cell.js:106]: This link seems not working.

File: packages/ui/Layout/Cell.js (hunk 10 above)
Comment (hoeunryuLG) [reply packages/ui/Layout/Cell.js:109]: I'll fix it.

File: packages/ui/Layout/Cell.js (hunk 11 above)
Comment (hoeunryuLG) [reply packages/ui/Layout/Cell.js:106]: I'll fix it.

**Review by seunghoh — APPROVED** [review]
LGTM

**Pull request merged**